src/main/java/org/opensearch/knn/index/KNNVectorDVLeafFieldData.java

```diff
@@ -48,6 +50,44 @@ public ScriptDocValues<float[]> getScriptValues() {
 
     @Override
     public SortedBinaryDocValues getBytesValues() {
-        throw new UnsupportedOperationException("knn vector field '" + fieldName + "' doesn't support sorting");
+        try {
+            final BinaryDocValues binaryDocValues = DocValues.getBinary(reader, fieldName);
+            SortedBinaryDocValues sortedBinaryDocValues = new SortedBinaryDocValues() {
+
+                private boolean docExists = false;
+                float[] floats = null;
+                int pos = 0;
+                BytesRefBuilder bytesRefBuilder = new BytesRefBuilder();
+
+                @Override
+                public boolean advanceExact(int doc) throws IOException {
+                    if (binaryDocValues.advanceExact(doc)) {
+                        docExists = true;
+                        floats = vectorDataType.getVectorFromDocValues(binaryDocValues.binaryValue());
+                        pos = 0;
+                        return docExists;
+                    }
+                    docExists = false;
+                    return docExists;
+                }
+
+                @Override
+                public int docValueCount() {
+                    return docExists ? floats.length : 1;
+                }
```

Comment:
what is the point of making this as floats.length? and if no doc exists then 1? can you please add some explanation here.